modules/phase_field/src/ics/TricrystalTripleJunctionIC.C

```diff
@@ -0,0 +1,128 @@
+}
+
+Real
+TricrystalTripleJunctionIC::value(const Point & p)
+{
+  /*
+   * This does all the work to create a triple junction that looks like the letter Y
+   */
+  Real dist_left; // The distance from the point to the line specified by _theta1
+  Real dist_right; // The distance from the point to the line specified by _theta2
+
+  // Check if the point is above or below the left-most line
+  // Function to use is y = max_y/2.0 + (x - max_x/2.0) * std::tan(libMesh::pi/2.0 - _theta1)
+  if (_theta1 == 0) // Handle tan(0) case
+    dist_left = p(1) - _range(1) / 2.0;
+  else
+    dist_left = p(1) - (_range(1)/2.0 + (p(0) - _range(0)/2.0) * _tan_theta1);
+
+  // Check if the point is above or below the right-most line
+  // Function to use is y = max_y/2.0 + (x - max_x/2.0)*std::tan(-(libMesh::pi/2.0 - _theta2))
+  if (_theta2 == 0) // Handle tan(0) classes
+    dist_right = p(1) - _range(1) / 2.0;
+  else
+    dist_right = p(1) - (_range(1)/2.0  + (p(0) - _range(0)/2.0) * _tan_theta2);
+
+  // Check if the point is to the left or right of the middle line
+  Real dist_center = p(0) - _range(0)/2.0; // Negative value if the point is to the left
+
+  if (_tan_theta1 > 0 && _theta1 <= libMesh::pi/2.0) // Case for large left grain
+  {
+    /*
+     * There's a lot going on here.  The first statement tells MOOSE to check and
+     * see if the current point is above the line defined by the first angle, only
+     * if it is past the center line.  All other points to the left of the center
+     * line are going to be part of the 1st grain (_op_index == 1)
+     * The second statement defines the second grain by the line defined by _theta2
+     * and marks everything below that line, and to the right of the center line.
+     * The third statement takes care of everything in between.
+     */
+    if ((((dist_left >= 0 && dist_center >= 0) || (dist_center < 0)) && _op_index == 1) ||
+        (dist_right <= 0 && dist_center > 0 && _op_index == 2) ||
+        (dist_left < 0 && dist_center > 0 && dist_right > 0 && _op_index == 3))
+      return 1.0;
+    else
+      return 0.0;
+  }
+
+  // This does a similar thing as the above case, but switches it for the right and left grains
+  else if (_tan_theta2 < 0 && _theta2 >= libMesh::pi/2.0) // Case for large right grain
+  {
+    if ((dist_left <= 0 && dist_center <= 0 && _op_index == 1) ||
+        (((dist_right >= 0 && dist_center <= 0) || (dist_center > 0)) && _op_index == 2) ||
+        (dist_left > 0 && dist_right < 0 && dist_center < 0 && _op_index == 3))
+      return 1.0;
+    else
+      return 0.0;
+  }
+
+  else // all other cases
+  {
+    if ((dist_left <= 0 && dist_center <= 0 && _op_index == 1) || // First grain
+        (dist_right <= 0 && dist_center > 0 && _op_index == 2) || // Second grain
+        (((dist_left > 0 && dist_center < 0) || (dist_right > 0 && dist_center >= 0)) && _op_index == 3)) // Third grain
+       return 1.0;
+    else
+       return 0.0;
```

Comment:
It would not be particularly hard to add the capability for smooth interfaces, but maybe we leave that for another PR.